Tests/iaas/flavor-naming/check-yaml.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""Check flavor names in the YAML files used for scs-0103-v1 with respect to scs-0100-vX.
+
+Also check that the flavor properties encoded in the YAML match those encoded in the name.
+"""
+import importlib
+import os
+import os.path
+import sys
+
+import yaml
+
+fnmck = importlib.import_module("flavor-name-check")
+fnmck.disallow_old = True
+
+
+REQUIRED_FIELDS = ['name-v1', 'name-v2', 'name', 'cpus', 'ram', 'cpu-type']
+DEFAULTS = {'disk0-type': 'network'}
+CPUTYPE_KEY = {'L': 'crowded-core', 'V': 'shared-core', 'T': 'dedicated-thread', 'C': 'dedicated-core'}
+DISKTYPE_KEY = {'n': 'network', 'h': 'hdd', 's': 'ssd', 'p': 'nvme'}
+
+
+class Undefined:
+    def __repr__(self):
+        return 'undefined'
+
+
+class Checker:
```

Comment:
opinion: can we add a docstring to this class?

Reply:
Yes, sure!